internal/heartbeat/heartbeat.go

```diff
@@ -231,35 +274,45 @@ func (s *Heartbeat) pushInformation() error {
 		Content:   content,
 		Directive: "heartbeat",
 	}
-
+	log.Debugf("pushInformation: sending content %+v; DeviceID: %s;", content, deviceId)
 	err = s.send(data)
 
+	s.pushInfoLock.Lock()
+	defer s.pushInfoLock.Unlock()
 	if err == nil {
 		s.firstHearbeat = false
 	} else if s.firstHearbeat {
-		s.data.previousMutableHardwareInfo = nil
+		s.data.SetPreviousHardwareInfo(nil)
```

Comment:
hmm. passing `nil`. IMO it was better before

Reply:
Yep I have to protect it, if I don't there may be a data race as the `pushInformation` method is called be the ticker goroutine so if for any reason, an new `initTicker` were called while a tick occurs, we could have simultaneous access
If the send fails, I have to reset the previous state if it was the 1st call otherwise, only delta will be sent in the heartbeat while the client never received the full first heartbeat as it should expect